graphql/schemabuilder/helpers.go

```diff
@@ -390,3 +405,111 @@ func castSlice(slice interface{}) []interface{} {
 
 	return ret
 }
+
+func (sb *schemaBuilder) buildPaginatedArgParser(originalArgType reflect.Type) (*argParser, graphql.Type, error) {
+
+	interfaceParser, interfaceType, err := sb.makeStructParser(originalArgType)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to build args for paginated field")
+	}
+
+	typ := reflect.TypeOf(ConnectionArgs{})
+	fields := make(map[string]argField)
+	argType := &graphql.InputObject{
+		Name:        typ.Name(),
+		InputFields: make(map[string]graphql.Type),
+	}
+	if argType.Name != "" {
+		argType.Name += "_InputObject"
+	}
+
+	if typ.Kind() != reflect.Struct {
+		return nil, nil, fmt.Errorf("expected struct but received type %s", typ.Name())
+	}
+	for i := 0; i < typ.NumField()-1; i++ {
```

Comment:
can we split out where this was copied from and reuse any of this? maybe into a helper function?